classes/manager.php

```diff
@@ -0,0 +1,570 @@
+        file_save_draft_area_files(
+            $data->compcatfiles,
+            $this->contextid,
+            'tiny_elements',
+            'images',
+            $data->id,
+            constants::FILE_OPTIONS
+        );
+
+        // Purge CSS cache if necessary.
+        if ($data->css != $oldrecord->css) {
+            \tiny_elements\local\utils::purge_css_cache();
+            \tiny_elements\local\utils::rebuild_css_cache();
+        }
+
+        $result = $DB->update_record(constants::TABLES['compcat'], $data);
+
+        $result &= $DB->execute(
+            'UPDATE {tiny_elements_component}
+             SET categoryname = ?
+             WHERE categoryname = ?',
+            [$data->name, $oldrecord->name]
+        );
+
+        $result &= $DB->execute(
+            'UPDATE {tiny_elements_flavor}
+             SET categoryname = ?
+             WHERE categoryname = ?',
+            [$data->name, $oldrecord->name]
+        );
+
+        $result &= $DB->execute(
+            'UPDATE {tiny_elements_variant}
+             SET categoryname = ?
+             WHERE categoryname = ?',
+            [$data->name, $oldrecord->name]
+        );
+
+        return $result;
+    }
+
+    /**
+     * Update a flavor.
+     *
+     * @param object $data
+     * @return bool
+     */
+    public function update_flavor(object $data): bool {
+        global $DB;
+        $data->timemodified = time();
+        $data->hideforstudents = !empty($data->hideforstudents);
+
+        $oldrecord = $DB->get_record(constants::TABLES['flavor'], ['id' => $data->id]);
+
+        // Purge CSS cache if necessary.
+        if ($data->css != $oldrecord->css) {
+            \tiny_elements\local\utils::purge_css_cache();
+            \tiny_elements\local\utils::rebuild_css_cache();
+        }
+
+        $result = $DB->update_record(constants::TABLES['flavor'], $data);
+
+        if ($oldrecord->name != $data->name) {
+            $result &= $DB->execute(
+                'UPDATE {tiny_elements_comp_flavor}
+                 SET flavorname = ?
+                 WHERE flavorname = ?',
+                [$data->name, $oldrecord->name]
+            );
+        }
+
+        return $result;
+    }
+
+    /**
+     * Update a variant.
+     *
+     * @param object $data
+     * @return bool
+     */
+    public function update_variant(object $data): bool {
+        global $DB;
+        $data->timemodified = time();
+
+        $oldrecord = $DB->get_record(constants::TABLES['variant'], ['id' => $data->id]);
+
+        // Purge CSS cache if necessary.
+        if ($data->css != $oldrecord->css || $data->iconurl != $oldrecord->iconurl) {
+            \tiny_elements\local\utils::purge_css_cache();
+            \tiny_elements\local\utils::rebuild_css_cache();
+        }
+
+        $result = $DB->update_record(constants::TABLES['variant'], $data);
+
+        if ($oldrecord->name != $data->name) {
+            $result &= $DB->execute(
```

Comment:
Using the return value from `$DB->execute(...)` makes sense, but `update_record` just always returns true, so does not really make sense to use the result from `update_record`. This probably occurs on some other places as well

Reply:
We have four places where code works like that (always having some subsequent $DB->execute() calls that are evaluated for result too - so we'd have to initialize $result with true (and the code does this ;-) ).
Another option would be to immediately query the updated record again and check whether the values are correct - but I don't think it's necessary.